include/conv2d.hpp

```diff
@@ -0,0 +1,88 @@
+#include "tensor.hpp"
+
+class Conv2D{
+    private:
+        Tensor kernel;
+        int padding;
+        int kernel_size;
+        int stride;
+        Tensor pad(Tensor &tensor);
+        std::string paddingMode;
+    public:
+        Conv2D(int kernel_size, int stride, int padding);
```

Comment:
We need to support, in_channels, out_channels. See example implementation https://github.com/ml-explore/mlx/blob/6589c869d6f2c0ba7c95fec50adfaf146f35d0b0/mlx/backend/common/conv.cpp